flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/utils/HeapMemorySegmentPool.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.utils;
+
+import org.apache.flink.core.memory.MemorySegment;
+import org.apache.flink.core.memory.MemorySegmentFactory;
+import org.apache.flink.table.runtime.util.MemorySegmentPool;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/** MemorySegment pool from heap. */
+public class HeapMemorySegmentPool implements MemorySegmentPool {
```

Comment:
Why do we need a new memory segment pool? Isn't `LazyMemorySegmentPool` sufficient?

Reply:
Sink does not support getting memory from `MemoryManager` now.